modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/item/selector/FDSViewItemSelector.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import {FrontendDataSet} from '@liferay/frontend-data-set-web';
+import React, {useState} from 'react';
+
+import {API_URL, PAGINATION_PROPS} from '../../js/Constants';
+
+import './FDSViewItemSelector.scss';
+
+const views = [
+	{
+		contentRenderer: 'list',
+		name: 'list',
+		schema: {
+			description: 'description',
+			symbol: 'symbol',
+			title: 'label',
+		},
+	},
+];
+
+const FDSViewItemSelector = ({
+	className,
+	classNameId,
+}: {
+	className: String;
+	classNameId: String;
+}) => {
+	const [classPK, setClassPK] = useState<String>();
+
+	return (
+		<div className="fds-view-item-selector">
+			<ClayModal.Body>
+				<FrontendDataSet
+					apiURL={API_URL.FDS_VIEWS}
+					id="FDSViewsItemSelector"
+					onSelect={({
+						selectedItems,
+					}: {
+						selectedItems: Array<{id: string}>;
+					}) => {
+						setClassPK(selectedItems[0].id);
+					}}
+					selectedItemsKey="id"
+					selectionType="single"
+					style="fluid"
+					views={views}
+					{...PAGINATION_PROPS}
```

Comment:
Do we want to put this before the other props so that we don't accidentally override the ones we have declared here?

Reply:
It's more or less the same, sets of props should never intersect. You would be very right if these were something like `DEFAULT_FDS_PROPS`, but these are just about pagination.

I considered at one point adding these as defaults in FDS itself, but that would mean possibly breaking some things in commerce, i.e. checking sages and migration.

Reply:
to avoid future bugs where there might be accidental overlap, can we move it above? More of a personal nit, I've been bitten by this a few times before where I can't figure out why my props aren't applying, only to realize it was because of where the destructure was happening. I prefer the "closest"(defined directly to the component) props applied to be the overriding ones.

Reply:
Yeah, I see your point. It does sound like a good practice. There is a lot of this in DS manager, we can make a small pet task to tweak all of them.